src/components/NanoContract/NanoContractTransactionActionListItem.component.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {
+  TouchableHighlight,
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { useSelector } from 'react-redux';
+import { t } from 'ttag';
+import { COLORS } from '../../styles/themes';
+import { getShortHash, isTokenNFT, renderValue } from '../../utils';
+import { ReceivedIcon } from '../Icons/Received.icon';
+import { SentIcon } from '../Icons/Sent.icon';
+
+/**
+ * It returns either 'sent' or 'received' depending on value.
+ *
+ * @param {number} value
+ *
+ * @returns {'sent'|'received'}
+ */
+function getBalanceType(value) {
+  if (value < 0) {
+    return 'sent';
+  }
+  return 'received';
+}
+
+/**
+ * Retrives token symbol, otherwise returns a shortened token hash.
+ *
+ * @param {string} tokenUid Token hash
+ * @param {Object[]} tokens Registered tokens from redux store
+ */
+function getTokenSymbol(tokenUid, tokens) {
+  const registeredToken = Object.values(tokens).filter((token) => token.uid === tokenUid).pop();
+  if (registeredToken) {
+    return registeredToken.symbol;
+  }
+  return getShortHash(tokenUid, 7);
+}
+
+/**
+ * It renders the item of Nano Contract Transactions List.
+ *
+ * @param {Object} props
+ * @param {Object} props.item registered Nano Contract data
+ * @param {number} props.index position in the list
+ */
+export const NanoContractTransactionActionListItem = ({ item, index }) => {
```

Comment:
It seems that this component has lot of mentions to balance, when it shouldn't. I prefer you don't add this component do this PR and we can add it once we implement the action list. What do you think?

Reply:
Well, we can do it for sure, but I have updated it now.